GDJS/Runtime/runtimeobject.js

```diff
@@ -34,27 +41,64 @@
  * @param {ObjectData} objectData The initial properties of the object.
  */
 gdjs.RuntimeObject = function(runtimeScene, objectData) {
+    /** @type {string} */
     this.name = objectData.name || "";
-    this._nameId = gdjs.RuntimeObject.getNameIdentifier(this.name);
+    /** @type {string} */
     this.type = objectData.type || "";
+    /** @type {number} */
     this.x = 0;
+    /** @type {number} */
     this.y = 0;
+    /** @type {number} */
     this.angle = 0;
+    /** @type {number} */
     this.zOrder = 0;
+    /** @type {boolean} */
     this.hidden = false;
+    /** @type {string} */
     this.layer = "";
-    this.livingOnScene = true;
-    this.id = runtimeScene.createNewUniqueId();
-    this._runtimeScene = runtimeScene; //This could/should be avoided.
+
+    /**
+     * @type {number}
+     * @protected
+     */
+    this._nameId = gdjs.RuntimeObject.getNameIdentifier(this.name);
+    /** 
+     * @type {boolean} 
+     * @protected
+     */
+    this._livingOnScene = true;
+    /**
+     * @type {number}
+     * @protected
+     */
+    this._id = runtimeScene.createNewUniqueId();
```

Comment:
This seems dangerous to refactor thits to _id in this PR. There is a regression in this file (see my other comment).
`id` is also from what I recall used sometimes outside of the class so I think if we want to refactor this, this has to be made very carefully (i.e: checking the whole extension and game engine for all `.id` references).

If there is just a single one `.id` that you forget to rename, this will be a regression and I may have to revert the whole PR => better make a PR first without this to get typings and *then* do the refactoring.

Typescript will help up to do so, by screaming in case something was forgotten to be renamed. But until we type most of the codebase, this is a very dangerous operation (I'm not even mentioning games that might use `id` already in their JS code).